packages/runtime-core/src/compat/componentFunctional.ts

```diff
@@ -35,7 +54,7 @@ export function convertLegacyFunctionalComponent(comp: ComponentOptions) {
       children: instance.vnode.children || [],
       data: instance.vnode.props || {},
       scopedSlots: ctx.slots,
-      parent: instance.parent && instance.parent.proxy,
+      parent: getFunctionalComponentParent(instance),
```

Comment:
I feel this is not a better fix.
Maybe we should set `instance.proxy` in https://github.com/vuejs/core/blob/e3bc21f39a684196af19d0ab187e03285c7377e1/packages/runtime-core/src/component.ts#L596
likely:
```javascript
if(!isStateful){
  instance.proxy=instance.parent.proxy
}
```

Reply:
Thanks a lot for reviewing!

Reply:
I am not sure.
lets waiting for others' input.